go/registry/api/api.go

```diff
@@ -305,12 +306,24 @@ type NodeUnfrozenEvent struct {
 
 // Event is a registry event returned via GetEvents.
 type Event struct {
+	TxHash hash.Hash `json:"tx_hash,omitempty"`
+
 	RuntimeEvent      *RuntimeEvent      `json:"runtime,omitempty"`
 	EntityEvent       *EntityEvent       `json:"entity,omitempty"`
 	NodeEvent         *NodeEvent         `json:"node,omitempty"`
 	NodeUnfrozenEvent *NodeUnfrozenEvent `json:"node_unfrozen,omitempty"`
 }
 
+// NodeListEpochInternalEvent is the per-epoch node list event.
+type NodeListEpochInternalEvent struct {
+	Height int64 `json:"height"`
+}
+
+// InternalEvent is a registry app internal event that is not exposed.
+type InternalEvent struct {
```

Comment:
Why even expose this in the general API? If it is internal then it is backend-specific so it should just be defined inside the backend implementation and not exposed.

Reply:
yeah agreed & fixed